tests/utilities/test_wheel.py

```diff
@@ -0,0 +1,162 @@
+import os
+import sys
+
+import pytest
+
+import tiledb
+import tiledb.cloud
+from tiledb.cloud.utilities import get_logger
+from tiledb.cloud.utilities import install_wheel
+from tiledb.cloud.utilities import upload_wheel
+from tiledb.cloud.utilities.wheel import PipInstall
+
+logger = get_logger()
+
+_LOCAL_WHEEL = "tests/utilities/data/fake_unittest_wheel-0.1.0-py3-none-any.whl"
+_ARRAY_NAME = os.path.basename(_LOCAL_WHEEL)
+_NAMESPACE = tiledb.cloud.client.default_user().username
+_S3_OBJECT_PATH = tiledb.cloud.client.default_user().default_s3_path
+_FULL_URI = os.path.join(
+    "tiledb://",
+    _NAMESPACE,
+    _S3_OBJECT_PATH,
+    _ARRAY_NAME,
+)
+_TDB_URI = os.path.join("tiledb://", _NAMESPACE, _ARRAY_NAME)
+_CONFIG = tiledb.cloud.Config()
+
+
+@pytest.fixture
+def array_teardown():
+    """Handle setup and teardown of arrays.
+
+    Ensures array doesn't exist and deletes if it does.
+    Deletes array after tests complete.
+    """
+
+    ctx = tiledb.Ctx(_CONFIG)
+
+    try:
+        tiledb.Array.delete_array(_FULL_URI, ctx=ctx)
+    except tiledb.TileDBError:
+        pass
+
+    yield None
+
+    tiledb.Array.delete_array(_FULL_URI, ctx=ctx)
+
+
+def test_upload_wheel(array_teardown) -> None:
+    # first time uploading array
+    upload_wheel(
+        wheel_path=_LOCAL_WHEEL,
+        dest_uri=_FULL_URI,
+        overwrite=True,
+        config=_CONFIG,
+    )
+
+    info = tiledb.cloud.info(os.path.join("tiledb://", _NAMESPACE, _ARRAY_NAME))
+
+    assert info.name == _ARRAY_NAME
+    assert info.namespace == _NAMESPACE
+    assert info.uri == os.path.join(_S3_OBJECT_PATH, _ARRAY_NAME)
+
+    with pytest.raises(ValueError):
+        upload_wheel(
+            wheel_path=_LOCAL_WHEEL,
+            dest_uri=_FULL_URI,
+            overwrite=False,
+            config=_CONFIG,
+        )
+
+    upload_wheel(
+        wheel_path=_LOCAL_WHEEL,
+        dest_uri=_FULL_URI,
+        overwrite=True,
+        config=_CONFIG,
```

Comment:
@spencerseale @gspowley @ryan-williams we're observing regular failures of this test in CI. Do we have a race condition in these uploads or something? Can we restructure the test to eliminate the race or collision?

Reply:
Yes, I believe there is a collision when more than one CI runner is running these tests at the same time. We can restructure the test to create unique asset names for each test.

Semi-related, there are 6 `run-tests` [runners](https://github.com/TileDB-Inc/TileDB-Cloud-Py/actions/runs/10709331229/job/29693713400) in GHA, which are now all running the all tests? In the past, I believe each runner ran a subset of the tests to reduce test time. All runners running the same tests would make the collision more likely.

We still need to restructure the test though.

Reply:
#636 modifies `test_wheel.py` to avoid the collision. Tests are currently in flight.

Reply:
Tests are passing in #636 now and the pytest splitting issue is resolved.

Reply:
@gspowley @sgillies sorry guys, I didn't check my notifications this week. George thank you for taking care of it.